src/dockerkeychain/docker_keychain.go

```diff
@@ -1,53 +1,84 @@
 package dockerkeychain
 
 import (
+	"bytes"
+	"context"
 	"sync"
 
 	"github.com/docker/cli/cli/config"
 	"github.com/docker/cli/cli/config/configfile"
 	dockertypes "github.com/docker/cli/cli/config/types"
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/google/go-containerregistry/pkg/name"
-	"github.com/spf13/afero"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type dockerKeychain struct {
-	mutex            sync.Mutex
-	dockerConfigFile string
-	filesystem       afero.Fs
+type DockerKeychain struct {
+	mutex        sync.Mutex
+	dockerConfig *configfile.ConfigFile
 }
 
-func NewDockerKeychain(dockerconfigFile string, filesystem afero.Fs) authn.Keychain {
-	return &dockerKeychain{
-		dockerConfigFile: dockerconfigFile,
-		filesystem:       filesystem,
+func NewDockerKeychain() *DockerKeychain {
+	keychain := &DockerKeychain{}
+	return keychain
+}
+
+func (keychain *DockerKeychain) LoadDockerConfigFromSecret(ctx context.Context, apiReader client.Reader, pullSecret corev1.Secret) error {
```

Comment:
is there a benefit to not just do this in the `NewDockerKeychain` ?

Reply:
I agree with @0sewa0 ☝️

Reply:
I did this for unit tests, but it is no longer required, so will change it back